modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -116,849 +132,590 @@ public function __construct()
-                        $pageData['additionalDetails'][$h] = '';
-                        $pageData['tag'][$h]['MAIN'] = 'a';
-                    }
+            if($folderObj->getType() === 'draft' && !array_key_exists('status', $filter_fields)) {
+                if (!empty($where)) {
+                    $where .= ' AND ';
+                }
+                $where .= ' emails.status LIKE "draft" ';
+            }
 
 
-                    // Filter imported emails based on the UID of the results from the IMAP server
-                    $crmWhere = $where . ' AND mailbox_id LIKE ' . '"' . $inboundEmail->id . '"';
-
-                    // Populates CRM fields
-                    $crmQueryArray = $seed->create_new_list_query(
-                        'id',
-                        $crmWhere,
-                        $filter_fields,
-                        $params,
-                        0,
-                        '',
-                        true,
-                        $seed,
-                        $singleSelect);
-
-                    $ret_array['inner_join'] = '';
-                    if (!empty($this->seed->listview_inner_join)) {
-                        $ret_array['inner_join'] = ' ' . implode(' ', $this->seed->listview_inner_join) . ' ';
-                    }
+            $folder = $inboundEmail->mailbox;
 
-                    if (!is_array($params)) {
-                        $params = array();
-                    }
-                    if (!isset($params['custom_select'])) {
-                        $params['custom_select'] = '';
-                    }
-                    if (!isset($params['custom_from'])) {
-                        $params['custom_from'] = '';
-                    }
-                    if (!isset($params['custom_where'])) {
-                        $params['custom_where'] = '';
-                    }
-                    if (!isset($params['custom_order_by'])) {
-                        $params['custom_order_by'] = '';
-                    }
+            $filter = $this->getFilter($filter_fields, $where, $request);
 
-                    $this->setVariableName($seed->object_name, $crmWhere, $this->listviewName, $id);
 
-                    $this->seed->id = '[SELECT_ID_LIST]';
+            // carry out the filter type
+            switch ($this->searchType) {
+                case 'crm':
 
-                    $nextOffset = -1;
-                    $prevOffset = -1;
-                    $endOffset = 0;
 
+                    $limit = $sugar_config['list_max_entries_per_page'];
 
-                    if ($total > $limitPerPage) {
-                        $nextOffset = $offset + $limitPerPage;
-                    }
+                    $search = new ListViewDataEmailsSearchOnCrm($this);
+                    $ret = $search->search($filter_fields, $request, $where, $inboundEmail, $params, $seed, $singleSelect, $id, $limit, $current_user, $id_field, $offset);
```

Comment:
Line goes over the 120 character limit. Consider splitting the into on line per argument.